crates/pathfinder/src/storage/schema/revision_0014.rs

```diff
@@ -0,0 +1,396 @@
+        pub contract_address: ContractAddress,
+        pub contract_address_salt: ContractAddressSalt,
+        // FIXME: why this has to be optional
+        pub class_hash: Option<ClassHash>,
+        #[serde_as(as = "Vec<ConstructorParamAsDecimalStr>")]
+        pub constructor_calldata: Vec<ConstructorParam>,
+        pub transaction_hash: StarknetTransactionHash,
+    }
+
+    /// Represents deserialized L2 invoke transaction data.
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+    pub struct InvokeTransaction {
+        #[serde_as(as = "Vec<CallParamAsDecimalStr>")]
+        pub calldata: Vec<CallParam>,
+        pub contract_address: ContractAddress,
+        pub entry_point_selector: EntryPoint,
+        pub entry_point_type: EntryPointType,
+        #[serde_as(as = "FeeAsHexStr")]
+        pub max_fee: Fee,
+        #[serde_as(as = "Vec<TransactionSignatureElemAsDecimalStr>")]
+        pub signature: Vec<TransactionSignatureElem>,
+        pub transaction_hash: StarknetTransactionHash,
+    }
+}
+
+pub(crate) fn migrate(transaction: &RusqliteTransaction<'_>) -> anyhow::Result<()> {
+    let todo: usize = transaction
+        .query_row("SELECT count(1) FROM starknet_transactions", [], |r| {
+            r.get(0)
+        })
+        .context("Count rows in starknet transactions table")?;
+
+    if todo == 0 {
+        return Ok(());
+    }
+
+    tracing::info!(
+        num_transactions=%todo,
+        "Decompressing and migrating transactions, this may take a while.",
+    );
+
+    let mut compressor = zstd::bulk::Compressor::new(1).context("Create zstd compressor")?;
+
+    let mut stmt = transaction
+        .prepare("SELECT hash, tx FROM starknet_transactions")
+        .context("Prepare transaction query")?;
+    let mut update_stmt = transaction
+        .prepare("UPDATE starknet_transactions SET tx = :tx WHERE hash = :hash")
+        .context("Prepare transaction update statement")?;
+    let mut query_class_hash_stmt = transaction
+        .prepare("SELECT hash FROM contracts WHERE address = :contract_address")
+        .context("Prepare class hash lookup statement")?;
+
+    let mut old_uncompressed_size = 0usize;
+    let mut old_compressed_size = 0usize;
+    let mut new_uncompressed_size = 0usize;
+    let mut new_compressed_size = 0usize;
+
+    let mut processed_rows = 0;
+    let batch_size = (todo / 11).max(100000);
+    let start_of_run = std::time::Instant::now();
+    let mut start_of_batch = start_of_run;
+
+    let mut rows = stmt.query([])?;
+    while let Some(r) = rows.next()? {
+        let transaction_hash = r.get_ref_unwrap("hash").as_blob()?;
+        let tx = r.get_ref_unwrap("tx").as_blob()?;
+
+        old_compressed_size += tx.len();
+        let tx = zstd::decode_all(tx).context("Decompressing transaction")?;
+        old_uncompressed_size += tx.len();
+        let tx: transaction::Transaction = serde_json::de::from_slice(&tx).context(format!(
```

Comment:
```suggestion
        let tx: transaction::Transaction = serde_json::from_slice(&tx).context(format!(
```

I don't understand where does the `::de` come from, it does haunt some of our files.